apollo-cache-interceptor/src/commonMain/kotlin/com/apollographql/apollo3/interceptor/cache/CacheExecutionContext.kt

```diff
@@ -7,18 +7,11 @@ import com.apollographql.apollo3.api.Operation
 import com.apollographql.apollo3.api.Response
 import com.apollographql.apollo3.cache.normalized.NormalizedCache
 import com.apollographql.apollo3.ApolloQueryRequest
+import com.apollographql.apollo3.api.RequestContext
+import com.apollographql.apollo3.api.ResponseContext
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 
-@ApolloExperimental
-data class CacheResponseExecutionContext(
-    val fromCache: Boolean
-) : ExecutionContext.Element {
-  override val key: ExecutionContext.Key<*> = Key
-
-  companion object Key : ExecutionContext.Key<CacheResponseExecutionContext>
-}
-
-enum class FetchPolicy {
+sealed class FetchPolicy : RequestContext(Key) {
```

Comment:
I went with the coroutines convention of not having a wrapper `Context` object but directly making the implementing class inherit from `Context`